kgforge/specializations/models/rdf/store_service.py

```diff
@@ -68,93 +71,95 @@ def resolve_context(self, iri: str) -> Dict:
+                    context, local_only=local_only
+                )
             except ValueError:
                 try:
-                    doc = self.context_store.service.resolve_context(context, local_only=False)
+                    doc = self.context_store.service.resolve_context(
+                        context, local_only=False
+                    )
                 except ValueError as e:
                     raise e
             document.update(self.recursive_resolve(doc))
         elif isinstance(context, dict):
             document.update(context)
         return document
 
-    def _load_shape(self, resource_id):
-        if resource_id not in self._imported:
-            try:
-                shape = self.context_store.retrieve(resource_id, version=None, cross_bucket=False)
-            except RetrievalError as e:
-                print(e, resource_id)
-                # failed, don't try to load again
-                self._imported.append(resource_id)
-            else:
-                json_dict = as_jsonld(shape, form="compacted", store_metadata=False, model_context=None,
-                                      metadata_context=None, context_resolver=self.context_store.service.resolve_context)
-                # this double conversion was due blank nodes were not "regenerated" with json-ld
-                temp_graph = Graph().parse(data=json.dumps(json_dict), format="json-ld")
-                self._graph.parse(data=temp_graph.serialize(format="n3"), format="n3")
-                self._imported.append(resource_id)
-                if hasattr(shape, "imports"):
-                    for dependency in shape.imports:
-                        self._load_shape(self.context.expand(dependency))
-
-    def _type_shape(self, iri: URIRef):
+    def load_shape_graph(self, graph_id: str, schema_id: str) -> Graph:
         try:
-            shape = self._sg.lookup_shape_from_node(iri)
-        except KeyError:
-            self._load_shape(self._shapes_to_resources[iri])
-            # reloads the shapes graph
-            self._sg = ShapesGraphWrapper(self._graph)
-            shape = self._sg.lookup_shape_from_node(iri)
-        return shape
+            schema_resource = self.context_store.retrieve(
+                schema_id, version=None, cross_bucket=False
+            )
+        except RetrievalError as e:
+            raise ValueError(f"Failed to retrieve {schema_id}: {str(e)}") from e
+        json_dict = as_jsonld(
+            schema_resource,
+            form="compacted",
+            store_metadata=False,
+            model_context=None,
+            metadata_context=None,
+            context_resolver=self.context_store.service.resolve_context,
+        )
+        # this double conversion was due blank nodes were not "regenerated" with json-ld
+        temp_graph = Graph().parse(data=json.dumps(json_dict), format="json-ld")
+        schema_graph = self._graph.graph(URIRef(graph_id))
```

Comment:
I wonder about this one: 
Since we're retrieving the schema, shouldn't we get rid of all the content of schema_graph inside of _graph? 

Especially with the transitive loading, schema_graph probably holds all the imports of this schema too. Might as well override the whole completely, instead of "merging", there could be outdated things?

Reply:
Very unlikely that something is in the schema_graph at this point before parsing. But I added schema_graph.remove((None, None, None)) to empty it before parsing.